This patch closes the system_info probe session, as soon as the sysinfo is
queried. This prevents system_info probes from hanging around till the process
completes.
This patch has not been contributed upstream, but is planned to be done by
 2013-Jul-12.

--- openscap-0.9.7/src/OVAL/oval_agent.c.~1~	2013-04-23 06:21:51.236000001 -0700
+++ openscap-0.9.7/src/OVAL/oval_agent.c	2013-06-13 18:01:31.297731735 -0700
@@ -107,6 +107,10 @@
 		oscap_free(ag_sess);
 		return NULL;
 	}
+#if	defined(__SVR4) && defined(__sun)
+	oval_probe_session_destroy(ag_sess->psess);
+	ag_sess->psess     = oval_probe_session_new(ag_sess->sys_model);
+#endif
 	oval_syschar_model_set_sysinfo(ag_sess->sys_model, sysinfo);
 	oval_sysinfo_free(sysinfo);
 
